### PR TITLE
fixed user pincode, added mqqt discovery for more data, other minor changes

### PR DESCRIPTION
**Pull request by pvtex**

### CODE REVIEW

**Comment by omersiar** [comment]
Thanks.  Much appreciated.

**Comment by matjack1** [comment]
Hey @pvtex thank you :) Can you please try rebasing these commits on top of `dev` to resolve the conflicts?

If you could also possibly reduce the number of commits it would help us better review the code. Otherwise just resolving the conflicts is fine :)

Thanks!

**Comment by matjack1** [comment]
@pvtex if you struggle with Git we can have a call and figure out things together, I'm very interested in having a look at your code. Feel free to send me an email if you want to go this way, you find the address in my Github profile. Cheers

**Comment by pvtex** [comment]
I am right now rebasing my reposotory to the device branch. But could take some time

**Comment by pvtex** [comment]
squashed all my commits and rebased my repo on dev branch

**Comment by matjack1** [comment]
Cool, thanks, can you please delete the .pio and ESP-Debug folders from your PR?

**Comment by pvtex** [comment]
deletet the folders .pio and ESP-Debug,
added them to .gitignore

File: Exception.txt
Comment (matjack1) [inline Exception.txt:1]: can you please delete this file?

File: src/websrc/index.html [hunk 1]
```diff
@@ -34,6 +34,7 @@
             <div class="sidebar-header">
                 <br>
                 <h1><i class="glyphicon glyphicon-tags" aria-hidden="true"></i>esp-rfid</h1>
+                <center<<p>RFID-Haustuer</p></div>center>
```
Comment (matjack1) [inline src/websrc/index.html]: can you please remove this?

File: src/websrc/js/esprfid.js [hunk 2]
```diff
@@ -2033,7 +2033,7 @@ function login() {
 
 function getLatestReleaseInfo() {
 
-  $.getJSON("https://api.github.com/repos/esprfid/esp-rfid/releases/latest").done(function(release) {
+  $.getJSON("https://api.github.com/repos/esp-rfid/esp-rfid/releases/latest").done(function(release) {
```
Comment (matjack1) [inline src/websrc/js/esprfid.js]: this should not be changed

**Comment by matjack1** [comment]
Sorry, I'm not getting what are you trying to achieve with this PR. Can you please provide a description of the change?

Also, it would be better to have one PR for one feature, to make it easier to review and discuss, so if there are multiple things here can we start from just one?

**Comment by pvtex** [comment]
mainly these things are changed/added:

- more mqtt data is send (pincode for example) and ha discovery for that added
  (easier to integrate in ha with alarmo, yu can use the pincode from esp-rfid for arm/disarm alarmo)
- systemname is shown on status page and in the menu under the esp-rfid-icon
   (makes it more easier to manage if you have more than one device)
- uptime on status page human readable see issue #607

File: src/mqtt.esp [hunk 3]
```diff
@@ -271,9 +320,18 @@ void mqttPublishAccess(time_t accesstime, String const &isknown, String const &t
...
 	root["username"] = user;
 	root["access"] = type;
-	root["time"] = accesstime;
+	
+	unsigned long offset_hours = 2;    // 2 hours for Europe/Berlin
+	unsigned long t_unix_date;
+	offset_hours = offset_hours * 3600;    // convert number of days to seconds
+ 	t_unix_date = accesstime + offset_hours;
+	char accesstimestr[9];
+	sprintf(accesstimestr, "%02d.%02d.%4d %02d:%02d:%02d\n", day(t_unix_date), month(t_unix_date), year(t_unix_date), hour(t_unix_date), minute(t_unix_date), second(t_unix_date));
```
Comment (matjack1) [inline src/mqtt.esp:332]: @pvtex can you please remove this time conversion to string? I agree that is more human readable, but we would change the MQTT API by doing this and it becomes more difficult to parse by machines

File: src/mqtt.esp [hunk 4]
```diff
@@ -316,8 +374,16 @@ void mqttPublishAccess(time_t accesstime, String const &isknown, int types[MAX_N
...
+	root["pincode"] = pincode;
 	root["username"] = user;
-	root["time"] = accesstime;
+	
+	unsigned long offset_hours = 2;    // 2 hours for Europe/Berlin
+	unsigned long t_unix_date;
+	offset_hours = offset_hours * 3600;    // convert number of days to seconds
+ 	t_unix_date = accesstime + offset_hours;
+	char accesstimestr[9];
+	sprintf(accesstimestr, "%02d.%02d.%4d %02d:%02d:%02d\n", day(t_unix_date), month(t_unix_date), year(t_unix_date), hour(t_unix_date), minute(t_unix_date), second(t_unix_date));
```
Comment (matjack1) [inline src/mqtt.esp:385]: can you please remove this as well? Same applies as above

File: src/websrc/index.html [hunk 5]
```diff
@@ -34,6 +34,7 @@
             <div class="sidebar-header">
                 <br>
                 <h1><i class="glyphicon glyphicon-tags" aria-hidden="true"></i>esp-rfid</h1>
+                <div id="systemname" style='text-align: center;'></div>
```
Comment (matjack1) [inline src/websrc/index.html:37]: can you please use `class="text-center"` instead of the inline style?

**Review by matjack1 — COMMENTED** [review]
Sorry, I just had time to give another (I think last!) round of comment, if you have time to have a look at this. Thanks!!

**Comment by matjack1** [comment]
I'm closing this since it's abandoned and I continue here: https://github.com/esprfid/esp-rfid/pull/654

**Pull request closed without merging**